sympy/combinatorics/homomorphisms.py

```diff
@@ -417,3 +419,88 @@ def block_homomorphism(group, blocks):
     images = {g: Permutation([b[p[i]^g] for i in identity]) for g in group.generators}
     H = GroupHomomorphism(group, codomain, images)
     return H
+
+def group_isomorphism(G, H):
+    '''
+    Compute an isomorphism between 2 given groups.
+
+    Arguments:
+        G (a finite `FpGroup` or a `PermutationGroup`) -- First group
+        H (a finite `FpGroup` or a `PermutationGroup`) -- Second group
+
+    Returns:
+    If the groups are not isomorphic, `None` is returned.
+    If the groups are isomorphic, the corresponding `GroupHomomorphism` object is returned.
+
+    Summary:
+    Uses the approach suggested by Robert Tarjan to compute the isomorphism between two groups.
+    First, the set of generators are mapped with the elements of `H` and we check if the mapping induces an isomorphism.
+
+    '''
+
+    if isinstance(G, FpGroup):
+        if G.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        G = G._to_perm_group()[0]
+    if isinstance(H, FpGroup):
+        if H.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        H = H._to_perm_group()[0]
```

Comment:
You could probably check at least if one group is infinite and the other isn't because then you know there is no isomorphism.

Reply:
Perhaps, in the case of infinite groups, you could also check if there is a map of the generators of one to the other s.t. the relators are the same, or if the group's relators are just in different order, because, for example,
```
>>> F, a, b = free_group("a b")
>>> S, c, d = free_group("c d")
>>> G = FpGroup(F, [a*b])
>>> H = FpGroup(S, [c*d])
```
or
```
>>> G = FpGroup(F, [a**2, b**3])
>>> H = FpGroup(F, [b**3, a**2])
```

then `G` and `H` are clearly isomorphic. (I'm actually wondering if in the first case SymPy will think they are the same group anyway because it treats all free groups on the same number of generators as the same so would have `F==H` - you could see if it does, in which case we don't need this case).

Reply:
Yes I can update that 👍